src/AuctionHouseBot.cpp

```diff
@@ -1584,6 +1584,101 @@ void AuctionHouseBot::InitializeConfiguration()
+        do
+        {
+            Field* fields = result->Fetch();
+            AuctionInfo ai = {
+                fields[0].Get<uint32>(),
+                fields[1].Get<uint32>(),
+                fields[2].Get<uint32>()
+            };
+            ahBotActiveAuctions.push_back(ai);
+        } while (result->NextRow());
+
+        // For each auction, refund bidder where possible, delete entry from AH
+        AuctionHouseObject* auctionHouse;
+        for (auto iter = ahBotActiveAuctions.begin(); iter != ahBotActiveAuctions.end(); ++iter)
+        {
+            AuctionInfo& ai = *iter;
+
+            // Get Auction House and Auction references
+            auctionHouse = nullptr;
+            switch (ai.houseID) {
+                case 2: auctionHouse = sAuctionMgr->GetAuctionsMap(AllianceConfig.GetAHFID()); break;
+                case 6: auctionHouse = sAuctionMgr->GetAuctionsMap(HordeConfig.GetAHFID()); break;
+                case 7: auctionHouse = sAuctionMgr->GetAuctionsMap(NeutralConfig.GetAHFID()); break;
+            }
+            if (auctionHouse == nullptr)
+                continue;
+
+            AuctionEntry* auction = auctionHouse->GetAuction(ai.itemID);
+            if (!auction)
+                continue;
+
+            // If auction has a bidder, refund that character
+            if (ai.characterGUID != 0)
+            {
+                // Lookup accountID and accountName associated with bidder character
+                std::string accountIDQueryString =   "SELECT account  FROM characters WHERE guid = {}";
+                std::string accountNameQueryString = "SELECT username FROM account    WHERE id   = {}";
+
+                QueryResult accountIDQueryResult = CharacterDatabase.Query(accountIDQueryString, ai.characterGUID);
+                if (!accountIDQueryResult)
+                    continue;
+
+                uint32 bidderAccountID = (*accountIDQueryResult)[0].Get<uint32>();
+
+                QueryResult accountNameQueryResult = LoginDatabase.Query(accountNameQueryString, bidderAccountID);
+                if (!accountNameQueryResult)
+                    continue;
+
+                std::string bidderAccountName = (*accountNameQueryResult)[0].Get<std::string>();
+
+                // Load Player information, and issue refund
+                auto session = std::make_unique<WorldSession>(
+                    bidderAccountID, std::move(bidderAccountName), 0, nullptr,
+                    SEC_PLAYER, sWorld->getIntConfig(CONFIG_EXPANSION), 0, LOCALE_enUS, 0, false, false, 0
+                );
+
+                auto player = std::make_unique<Player>(session.get());
+                if (!player)
+                    continue;
+
+                player->Initialize(ai.characterGUID);
+
+                sAuctionMgr->SendAuctionCancelledToBidderMail(auction,  trans);
+                player->ModifyMoney(-int32(auction->GetAuctionCut())); // ModifyMoney() already handles negative check
```

Comment:
This isn't necessary / Is incorrect here.

The ->ModifyMoney line found in AuctionHouseHandler is for the one who put the auction up, not the one who bid on the auction.  The bidder gets their money back from the mail attachment in "SendAuctionCancelledToBidderMail", so this double dips.  And we wouldn't want to just switch this player to the ahbot because we don't care if the ahbot gets their deposit back.